cdc/sink/producer/kafka/metrics.go

```diff
@@ -0,0 +1,197 @@
+				Subsystem: "sink",
+				Name:      "kafka_producer_batch_size",
+				Help:      "the number of bytes sent per partition per request for all topics",
+			}, []string{"capture", "changefeed"}),
+	}
+
+	// `Meter`
+	recordSendRate = rawSaramaMetric{
+		metricName: "record-send-rate",
+		collector: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "ticdc",
+				Subsystem: "sink",
+				Name:      "kafka_producer_record_send_rate",
+				Help:      "Records/second sent to all topics",
+			}, []string{"capture", "changefeed"}),
+	}
+
+	// `Histogram`
+	recordsPerRequest = rawSaramaMetric{
+		metricName: "records-per-request",
+		collector: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "ticdc",
+				Subsystem: "sink",
+				Name:      "kafka_producer_records_per_request",
+				Help:      "the number of records sent per request for all topics",
+			}, []string{"capture", "changefeed"}),
+	}
+
+	// `Histogram`
+	compressionRatio = rawSaramaMetric{
+		metricName: "compression-ratio",
+		collector: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "ticdc",
+				Subsystem: "sink",
+				Name:      "kafka_producer_compression_ratio",
+				Help:      "the compression ratio times 100 of record batches for all topics",
+			}, []string{"capture", "changefeed"}),
+	}
+)
+
+// InitMetrics registers all metrics in this file
+func InitMetrics(registry *prometheus.Registry) {
+	registry.MustRegister(batchSize.collector)
+	registry.MustRegister(recordSendRate.collector)
+	registry.MustRegister(recordsPerRequest.collector)
+	registry.MustRegister(compressionRatio.collector)
+}
+
+type saramaMetricsMonitor struct {
+	captureAddr  string
+	changefeedID string
+
+	registry metrics.Registry
+	metrics  []saramaMetric
+}
+
+// Refresh all monitored metrics
+func (sm *saramaMetricsMonitor) Refresh() {
+	for _, m := range sm.metrics {
+		m.refresh(sm.registry)
+	}
+}
+
+func NewSaramaMetricsMonitor(registry metrics.Registry, captureAddr, changefeedID string) *saramaMetricsMonitor {
+	metrics := make([]saramaMetric, 0)
+	metrics = append(metrics, batchSize.withLabelValues(captureAddr, changefeedID))
+	metrics = append(metrics, recordSendRate.withLabelValues(captureAddr, changefeedID))
+	metrics = append(metrics, recordsPerRequest.withLabelValues(captureAddr, changefeedID))
+	metrics = append(metrics, compressionRatio.withLabelValues(captureAddr, changefeedID))
+
+	return &saramaMetricsMonitor{
+		captureAddr:  captureAddr,
+		changefeedID: changefeedID,
+		registry:     registry,
+		metrics:      metrics,
+	}
+}
+
+func (sm *saramaMetricsMonitor) Cleanup() {
+	batchSize.deleteLabelValues(sm.captureAddr, sm.changefeedID)
+	recordSendRate.deleteLabelValues(sm.captureAddr, sm.changefeedID)
+	recordsPerRequest.deleteLabelValues(sm.captureAddr, sm.changefeedID)
+	compressionRatio.deleteLabelValues(sm.captureAddr, sm.changefeedID)
+}
+
+// rawSaramaMetric is prometheus metric without label values.
+type rawSaramaMetric struct {
```

Comment:
I think we can simplify the implementation by remove the struct, like https://github.com/pingcap/tiflow/blob/e2d529cc9887ab61063555dbbbf1d9f9765581e3/pkg/db/leveldb.go#L111-L142